core/tests/webdriverio_utils/ContributorDashboardAdminPage.js

```diff
@@ -232,7 +236,123 @@ var ContributorDashboardAdminPage = function () {
       `.e2e-test-language-selector-option=${language}`
     );
     await action.click(`${language} option selector`, selectorOption);
-    await action.click('Language Selector', languageSelector);
+  };
+
+  this.selectDate = async function (
+    datePickerToggle,
+    datePickerInput,
+    selectedDate
+  ) {
+    var initialValueOfDatePicker = await action.getValue(
+      'Date Picker Input',
+      datePickerInput
+    );
+
+    await action.click('Date Picker Toggle', datePickerToggle);
+
+    // Here selectedDate is a Date object that is to be selected by date picker.
+    var day = selectedDate.getDate();
+    var month = selectedDate.getMonth() + 1;
+    var year = selectedDate.getFullYear();
+
+    if (year) {
+      var nextMonthButton = $(
+        '//body/div[1]/div[2]/div/mat-datepicker-content/div[2]/' +
+          'mat-calendar/mat-calendar-header/div/div/button[3]'
+      );
+      var prevMonthButton = $(
+        '//body/div[1]/div[2]/div/mat-datepicker-content/div[2]/' +
+          'mat-calendar/mat-calendar-header/div/div/button[2]'
+      );
+
+      var yearsToNavigate =
+        year - new Date(initialValueOfDatePicker).getFullYear();
+
+      if (yearsToNavigate > 0) {
+        for (let i = 0; i < yearsToNavigate * 12; i++) {
+          await waitFor.visibilityOf(
+            nextMonthButton,
+            'Next Month Button is not visible'
+          );
```

Comment:
I don't think this is quite right. Suppose the next calendar takes a long time to advance to the next month and stays stuck on the previous month for a while. The next month button will still be visible, but it'll be on the previous month. We need to wait for the next month to appear and click that month's next button, not the next button of the original month. The current code just checks that the next button is visible, not that it's the next button of the appropriate month.

Does this make sense? Please let me know if I'm misunderstanding something about how this works.

Ditto for the negative case below